cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -493,5 +496,27 @@ func main() {
 		}),
 		Conns:              connset,
 		RefreshCfgThrottle: refreshCfgThrottle,
-	}).Run(connSrc)
+	}
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGTERM)
+
+	go func() {
+		<-signals
+		logging.Infof("Received TERM signal. Waiting up to %s seconds before terminating.", *termTimeout)
```

Comment:
echoing easwars comment, it looks like you want to have a line like `proxyClient.MaxConnections = 0` here in order to stop accepting new connections